app/chat/ls.go

```diff
@@ -334,3 +337,216 @@ func applyPeers(ctx context.Context, manager *peers.Manager, entities peer.Entit
+			break
+		}
+
+		var (
+			dialogsSlice []tg.DialogClass
+			messages     []tg.MessageClass
+			users        []tg.UserClass
+			chats        []tg.ChatClass
+		)
+
+		switch d := result.(type) {
+		case *tg.MessagesDialogs:
+			dialogsSlice = d.Dialogs
+			messages = d.Messages
+			users = d.Users
+			chats = d.Chats
+		case *tg.MessagesDialogsSlice:
+			dialogsSlice = d.Dialogs
+			messages = d.Messages
+			users = d.Users
+			chats = d.Chats
+		case *tg.MessagesDialogsNotModified:
+			// No more dialogs
+			return allElems, skipped
+		default:
+			log.Error("unexpected dialog type", zap.String("type", fmt.Sprintf("%T", result)))
+			return allElems, skipped
+		}
+
+		if len(dialogsSlice) == 0 {
+			break
+		}
+
+		// Build entities map for this batch
+		// Convert slices to maps as required by peer.NewEntities
+		userMap := make(map[int64]*tg.User)
+		for _, u := range users {
+			if user, ok := u.(*tg.User); ok {
+				userMap[user.ID] = user
+			}
+		}
+
+		chatMap := make(map[int64]*tg.Chat)
+		channelMap := make(map[int64]*tg.Channel)
+		for _, c := range chats {
+			switch chat := c.(type) {
+			case *tg.Chat:
+				chatMap[chat.ID] = chat
+			case *tg.Channel:
+				channelMap[chat.ID] = chat
+			}
+		}
+
+		entities := peer.NewEntities(userMap, chatMap, channelMap)
+
+		// Process each dialog in this batch
+		for _, d := range dialogsSlice {
+			dialog, ok := d.(*tg.Dialog)
+			if !ok {
+				continue
+			}
+
+			// Find the peer ID for logging purposes
+			var peerID int64
+			switch p := dialog.Peer.(type) {
+			case *tg.PeerUser:
+				peerID = p.UserID
+			case *tg.PeerChat:
+				peerID = p.ChatID
+			case *tg.PeerChannel:
+				peerID = p.ChannelID
+			default:
+				log.Error("unknown peer type", zap.String("type", fmt.Sprintf("%T", p)))
+				skipped++
+				continue
+			}
+
+			// Skip if we've already seen this dialog (deduplication)
+			if seen[peerID] {
+				continue
+			}
+			seen[peerID] = true
+
+			// Try to extract the peer - THIS IS WHERE THE ORIGINAL ERROR OCCURS
+			// In gotd's query/dialogs iterator, it calls ExtractPeer without error handling,
+			// causing a panic when a channel doesn't exist in entities.
+			// We catch it here and skip the problematic dialog instead.
+			// See: https://github.com/iyear/tdl/issues/713
+			inputPeer, err := entities.ExtractPeer(dialog.Peer)
+			if err != nil {
+				// This dialog references a channel/chat that doesn't exist in entities
+				// (likely deleted, user was banned, or channel is inaccessible).
+				// Log and skip it instead of failing.
+				log.Warn("skipping dialog with missing peer",
+					zap.Int64("peer_id", peerID),
+					zap.String("peer_type", fmt.Sprintf("%T", dialog.Peer)),
+					zap.Error(err))
+				skipped++
+				continue
+			}
+
+			// Find the last message for this dialog
+			var lastMsg tg.NotEmptyMessage
+			for _, msg := range messages {
+				switch m := msg.(type) {
+				case *tg.Message:
+					if m.ID == dialog.TopMessage {
+						lastMsg = m
+						break
```

Comment:
This break only break switch not entire for loop.
Maybe use an inline func here.